src/util/cmdlineargs.h

```diff
@@ -11,6 +11,8 @@
 // A structure to store the parsed command-line arguments
 class CmdlineArgs final {
   public:
+    CmdlineArgs();
```

Comment:
Please add a remark that the default constructor is only supposed to be used in tests and that all operational code must access the singleton via Instance().